back/routes/review.js

```diff
@@ -0,0 +1,105 @@
+require("dotenv/config")
+var express = require('express');
+var router = express.Router();
+const { Parse } = require('./../parse')
+
+// ----- Get Reviews for an Experience -----
+router.post('/experience/reviews', async(req, res) => {
+  const query = new Parse.Query('Review');
+  query.equalTo("experienceId", req.body.experienceId);
+  let experienceReviews = await query.find();
+  res.status(200).send(experienceReviews);
+})
+
+// Find review of a user X to experience Y
+async function getReview(username, experienceId) {
+  const query = new Parse.Query('Review');
+  query.equalTo('adventurerUsername', username);
+  query.equalTo('experienceId', experienceId);
+  return query.first()
+}
+
+// ----- Check if an user has rated that experience -----
+router.post('/isRated', async(req, res) => {
+  let review = await getReview(req.body.username, req.body.experienceId)
+  if (review) {
+    res.status(200).send(true)
+  } else {
+    res.status(200).send(false);
+  }
+})
+
+// ----- Submit a review -----
+router.post('/rate', async(req, res) => {
+  let hasError = false
+  await Promise.all((req.body.reviews).map(async (adventurerReview) => {
+    const review = new Parse.Object('Review');
+    review.set('feedbackId', adventurerReview.feedbackId)
+    review.set('adventurerUsername', req.body.username)
+    review.set('experienceId', req.body.experienceId)
+    review.set('score', adventurerReview.score)
+    review.set('comment', adventurerReview.comment)
+    try {
+      await review.save();
+    } catch(err) {
+      hasError = true
+    }
+  }))
+  if (hasError) {
+    res.status(400).send(false);
+  }
+  else {
+    res.status(200).send(true);
+  }
+})
+
+// Query for all reviews
+async function getReviews(experienceId) {
+  const query = new Parse.Query('Review');
+  query.equalTo("experienceId", experienceId);
+  const reviews = await query.find();
+  return reviews.map((review) => review.toJSON())
+}
+
+// ----- Receive ALL reviews according to experienceId -----
+router.post('/all', async(req, res) => {
+  const reviews = await getReviews(req.body.experienceId)
+  res.status(200).send(reviews);
+})
+
+// Query for active feedback areas 
+async function activeFeedbackQuery(objectId){
+  const query = new Parse.Query('Experience');
+  query.equalTo('objectId', objectId);
+  let experienceInfo = await query.first();
+  if (experienceInfo) {
+    return experienceInfo.toJSON().activeFeedback
+  }
```

Comment:
Something like this might work?
```suggestion
  let experienceInfo = await query.first().select('activeFeedback');
```

Reply:
That is perfect! I didn't know there was a select option!
I was already thinking about doing a query document, and in those queries function add the .JSON. But this is way better.
I am thinking of not going (at least rn) for this implementation since I will take your other advice on making general requests and not having several for specific information. 
After refactoring, I will probably go back to this comment